plugins/flashrom/fu-flashrom-device.c

```diff
@@ -21,6 +21,8 @@ fu_flashrom_device_init (FuFlashromDevice *self)
 	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_INTERNAL);
 	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_UPDATABLE);
 	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_REQUIRE_AC);
+	fu_device_add_flag (FU_DEVICE (self), FWUPD_DEVICE_FLAG_USABLE_DURING_UPDATE);
```

Comment:
I suppose the latter is true, indeed.

Reply:
I think it's true for coreboot devices with multiple partitions, but all flashrom devices?  I don't know.

Reply:
I don't know of any machines that explode when you write to the SPI; some machines prevent writing the ME but flashrom won't try to (or be able to) write the SPI area in that case.

Reply:
The BIOS code i'm not concerned about.  Ignoring ME - I think it comes down to whether or not the EC stores an image on the SPI and if that is being updated at the same time and has multiple banks and how it gets restarted.

I would feel more comfortable if this flag was added only on a HardwareId to match the expectations for a manufacturer and not generic.

Reply:
> I would feel more comfortable if this flag was added only on a HardwareId to match the expectations for a manufacturer and not generic.

Works for me; @Sean-StarLabs I think you know what to do.

Reply:
@superm1 EC does store image on the SPI but it behaves the same way as SuperIO, that it doesn't take effect until reboot. No problems with removing the flag or making it specific, but I'm interested in what you think would go wrong?

Reply:
If an external MCU such as an ME or EC is actively using the SPI chip at the same time as OS process you can get a race condition or corruption on the SPI.  This is why for other plugins we have things such as Synaptics MST will stop HDCP and set up quad mode before flashing.